### PR TITLE
Add mutli ack to AckHandler

### PR DESCRIPTION
**Pull request by xingsy97**
1. Add multiple ack for `AckHandler`.
2. Refactor it

### CODE REVIEW

**Comment by vicancy** [comment]
> Refactor it to make it consistent with [RT version]

Is this a breaking change from SDK's perspective of view?

**Comment by xingsy97** [comment]
> > Refactor it to make it consistent with [RT version]
> 
> Is this a breaking change from SDK's perspective of view?

`AckHandler` is an internal class. I think it's not.

**Comment by vicancy** [comment]
>While SDK AckHandler will set its result with AckStatus.Timeout rather than cancelling it

So now ackHandler callers needs to catch the CancelldException?

**Comment by xingsy97** [comment]
> > While SDK AckHandler will set its result with AckStatus.Timeout rather than cancelling it
> 
> So now ackHandler callers needs to catch the CancelldException?

No, In SDK, AckHandler won't cancel the task when timeout. So catching `CancelldException` is not needed. It will set a result whose value is `AckStatus.Timeout`

File: src/Microsoft.Azure.SignalR.Common/Utilities/AckHandler.cs [hunk 1]
```diff
@@ -56,62 +54,228 @@ public static bool HandleAckStatus(IAckableMessage message, AckStatus status)
...
+            // `foreach (var (id, ack) in _acks)` cannot pass under netstandard2.0
+            foreach (var item in _acks)
+            {
+                var id = item.Key;
+                var ack = item.Value;
+                if (utcNow > ack.TimeoutAt)
+                {
+                    if (_acks.TryRemove(id, out _))
+                    {
+                        // This part is slightly different from RT.
```
Comment (vicancy) [inline src/Microsoft.Azure.SignalR.Common/Utilities/AckHandler.cs:128]: remove RT related comments

**Review by vicancy — APPROVED** [review]

**Pull request merged**